### ISSUE TITLE
Semver / versioning

### ISSUE THREAD

**Issue by RangerMauve** [benwiley4000/dat-pkg#1]
I think that there should be a folder for the actual package contents and a folder for metadata for the versions.

When you tag a new version you should write a file for the version name that points to a Dat url with a version in it so that people can easily checkout that version.

E.g.

If I have a file in `/versions/4.2.0` which points to `dat://mykey+666/data` and I do `npm install something@4`, it'll search the versions for the latest file that matches `@4`, checkout the Dat, and sync it's data to the FS.

**Comment by RangerMauve** [comment]
Alternatively we'd need tagging at the hyperdrive level. 🤷‍♂️

**Comment by benwiley4000** [comment]
I was kind of hoping that tagging at the hyperdrive level would exist but I guess it doesn't! This is a good idea.. we'll need some kind of enforcement that version tags can't change and that they need to be committed as part of the commit they reference. Which... Actually is that even possible to enforce? I'm not good enough at math to figure that out off the top of my head. Is it possible to get the commit id before the commit is made, and include it in the contents of the commit?

**Comment by benwiley4000** [comment]
If we remove the requirement that the tag is included with the commit itself, it's easier (and maybe that's fine)

**Comment by RangerMauve** [comment]
There's no commits in Dat. :O every change is a separate version in the metadata log. I was thinking that the head of the archive is `@latest` and might be unstable, and you add a new version by adding a version file to the versions folder

**Comment by benwiley4000** [comment]
Ok whatever the terminology does what I said basically make sense?

**Comment by RangerMauve** [comment]
Yes. If we consider commits to be the files I'm the versions folder then we could take the current "version number" of the Dat at the time.

**Comment by benwiley4000** [comment]
I made some updates to the readme partially based on this issue.. feel free to make any suggestions of your own.

**Comment by RangerMauve** [comment]
Yeah, I like it! I think regarding the build step, we should reuse the logic from package.json

**Comment by benwiley4000** [comment]
What do you mean.. like the prepare script that gets run on install when `npm install` is run inside a package repo?

**Comment by RangerMauve** [comment]
Yeah

**Comment by RangerMauve** [comment]
I think reusing as much as possible from NPM would be a good approach.

**Comment by benwiley4000** [comment]
So I guess instead of specifying a build command, we just go to that repo and run `npm install` and then compare the result?

**Comment by RangerMauve** [comment]
Err, for verification? I think I'm a little lost. 😵 I thought having it in the archive would be enough for verification since the owner controls it already.
